random-reviewer-picker/main.js

```diff
@@ -0,0 +1,134 @@
+const github = require("@actions/github");
+const core = require("@actions/core");
+const { random } = require("lodash");
+
+/**
+ * Fetch and transform inputs into usable variables
+ **/
+function getInputs() {
+  const reviewer_list = core
+    .getInput("reviewer_list")
+    .split(",")
+    .map(i => i.replace(" ", ""));
+  const reviewer_amount = parseInt(core.getInput("reviewer_amount"));
+  const maintainer_list = core
+    .getInput("maintainer_list")
+    .split(",")
+    .map(i => i.replace(" ", ""));
+  const maintainer_amount = parseInt(core.getInput("maintainer_amount"));
+  const skip_busy = !!core.getInput("skip_busy");
+
+  const GITHUB_TOKEN = core.getInput("GITHUB_TOKEN");
+
+  return {
+    reviewer_list,
+    reviewer_amount,
+    maintainer_list,
+    maintainer_amount,
+    skip_busy,
+    GITHUB_TOKEN
+  };
+}
+
+/**
+ * Get user "busy" status from github API
+ **/
+async function isUserBusy(octokit, userHandle) {
+  const { userStatus } = await octokit.graphql({
+    query: `
+    query userStatus($user: String!) {
+      user(login: $user}) {
+        status {
+          indicatesLimitedAvailability
+        }
+      }
+    }`,
+    user: userHandle
+  });
+
+  return userStatus.status.indicatesLimitedAvailability;
+}
+
+/**
+ * Main action entry point
+ **/
+async function run() {
+  //Get action events
+  const actionEvents = require(process.env.GITHUB_EVENT_PATH);
+
+  const {
+    reviewer_list,
+    reviewer_amount,
+    maintainer_list,
+    maintainer_amount,
+    skip_busy,
+    GITHUB_TOKEN
+  } = getInputs();
+
+  const [owner, repo] = process.env.GITHUB_REPOSITORY.split("/", 2);
+  const pull_number = actionEvents.pull_request.number;
+
+  const octokit = new github.GitHub(GITHUB_TOKEN);
```

Comment:
I understand why name octokit but in this place it's only a GitHub Client for us.
Would you prefer it to be githubClient only ?

Reply:
It's the actual name of the client. Their examples and documentations uses that name and I think most developers kept it, I continued that for consistancy

Reply:
Sure, if that sounds reasonable :)